AppSyncRealTimeClient/ConnectionProvider/AppsyncRealtimeConnection/RealtimeConnectionProvider+MessageInterceptable.swift

```diff
@@ -13,7 +13,46 @@ extension RealtimeConnectionProvider: MessageInterceptable {
         messageInterceptors.append(interceptor)
     }
 
+    public func interceptMessage(
+        _ message: AppSyncMessage,
+        for endpoint: URL,
+        completion: @escaping (AppSyncMessage) -> Void
+    ) {
+
+        chainInterceptors(
+            index: 0,
+            message: message,
+            endpoint: endpoint,
+            completion: completion
+        )
+    }
+
+    private func chainInterceptors(
+        index: Int,
+        message: AppSyncMessage,
+        endpoint: URL,
+        completion: @escaping (AppSyncMessage) -> Void
+    ) {
+
+        guard index < messageInterceptors.count else {
+            completion(message)
+            return
+        }
+        let interceptor = messageInterceptors[index]
```

Comment:
When multiple threads are executing concurrently, attempting to check the index boundary and retrieve an item using an index may not always be successful if another thread is also modifying the `messageInterceptors`.

Maybe use `Iterator.next()`:

```swift
public func interceptMessage(
        _ message: AppSyncMessage,
        for endpoint: URL,
        completion: @escaping (AppSyncMessage) -> Void
    ) {

        chainInterceptors(
            iterator: messageInterceptors.makeIterator(),
            message: message,
            endpoint: endpoint,
            completion: completion
        )
    }

    private func chainInterceptors(
        iterator: IteratorProtocol,
        message: AppSyncMessage,
        endpoint: URL,
        completion: @escaping (AppSyncMessage) -> Void
    ) {

        guard let interceptor = iterator.next() else {
            completion(message)
            return
        }
        ...
    }
```

Reply:
Ideally this should not happen in our code logic. But added the recommendation to be on the safer side.